nodejs/week2/app.js

```diff
@@ -0,0 +1,56 @@
+const { json } = require("express");
+const express = require("express");
+const app = express();
+const port = process.env.PORT || 3000;
+const documents = require("./documents.json");
+
+// Support parsing JSON requests
+app.use(express.json());
+
+app.get("/search", (req, res) => {
+  console.log(JSON.stringify(req.query));
+  const query = req.query.q.toLowerCase();
+  if (!query) {
+    res.json(documents);
+  } else {
+    const filtereddocs = documents.filter((document) => {
+      const matchesValue =
+        document.value && document.value.toLowerCase().includes(query);
+      const matchesName =
+        document.name && document.name.toLowerCase().includes(query);
+      const matchDesc =
+        document.description &&
+        document.description.toLowerCase().includes(query);
+      return matchesName || matchDesc || matchesValue;
```

Comment:
Here you are looking for specific fields (value, name and description) but the idea was for the search engine to support any field name. That would you should loop over each value in a document. One way of doing that is with `for...in`: [see docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in)